docs/design/specs/PortablePdb-Metadata.md

```diff
@@ -523,3 +525,75 @@ Parent: Module
+| mvid       | GUID (16 bytes)                           | Module Version Id (ModuleDef table field).                |
+
+The meaning of the _flags_ byte:
+
+| flag      | description                                                     |
+|:----------|:----------------------------------------------------------------|
+| 0b0000001 | Embed interop types.                                            |
+| 0b0000010 | The referenced file is an assembly (as opposed to a netmodule). |
+
+The remaining bits are reserved for future use and have currently no meaning.
+
+> The data can be used to find the reference in a file indexing service such as a symbol server. 
+> For example, the [Simple Symbol Query Protocol](https://github.com/dotnet/symstore/blob/master/docs/specs/Simple_Symbol_Query_Protocol.md) uses a combination of _file-name_, _time-stamp_ and _file-size_ as a [key](https://github.com/dotnet/symstore/blob/master/docs/specs/SSQP_Key_Conventions.md#pe-timestamp-filesize).
+> Other services might use the MVID as it uniquely identifies the module.
+
+##### <a name="CompilationOptions"></a>Compilation Options (C# and VB compilers)
+Parent: Module
+
+Kind: {B5FEEC05-8CD0-4A83-96DA-466284BB4BD8}
+
+Stores compilation options used to compile the module. Only captures information that is not present elsewhere in the PDB, in the PE headers or metadata of the module.
+
+The blob has the following structure:
+
+    Blob ::= (name value)*
+
+| terminal   | encoding            | description                      |
+|:-----------|:--------------------|:---------------------------------|
+| name       | UTF8 NIL-terminated | Name of the compilation option.  |
+| value      | UTF8 NIL-terminated | Value of the compilation option. |
+
+There shall be no two entries with the same _name_ in the list.
+
+It is recommended, but not required that _name_ is lower-case and uses hyphen (`-`) for separating words.
+
+Common options:
+
+| name               | value format                                                   | description                          |
+|:-------------------|:---------------------------------------------------------------|:-------------------------------------|
+| `language`         | `CSharp` or `VisualBasic`                                      | Language name.                       |
+| `language-version` | `[0-9]+(\.[0-9]+)?`                                            | Language version.                    |
+| `compiler-version` | [SemVer2](https://semver.org/spec/v2.0.0.html) version string  | Version of the compiler used to build the module with _build metadata_ set to commit SHA for officially released compiler. |
+| `runtime-version`  | [SemVer2](https://semver.org/spec/v2.0.0.html) version string  | Version of the CLR used to build the module with _build metadata_ set to commit SHA for officially released .NET Core runtime. |
```

Comment:
Is this supposed to be an exhaustive list? Missing: 

`strict`
`checked`
`define`
`optimization`

sharp has: 

`nullable`
`unsafe`
`default-encoding`
`fallback-encoding`
`portability-policy`

Reply:
No, this is not supposed to list C# or VB specific options as this is not a compiler spec. 

> Other options listed in the blob are specific to each compiler. Future versions of the compiler may add additional options.

Reply:
The idea here is to list general options that allow one to write a tool that finds the compiler and the runtime to use. Then the tool needs to have special code for each supported language.

Reply:
I guess maybe we should actually remote `language-version` since that might be too specific.

Reply:
If F# implemented this feature as well we would add `language` = `FSharp` and it would also have `compiler-version` and `runtime-version` but everything else would be F# specific.